core-spi-test/src/main/java/org/ehcache/internal/store/StoreClearTest.java

```diff
@@ -67,10 +69,12 @@ public void removesAllOfTheMappings()
     try {
       kvStore.clear();
     } catch (CacheAccessException e) {
-      System.err.println("Warning, an exception is thrown due to the SPI test");
-      e.printStackTrace();
+      validException = true;
     }
 
     assertThat(kvStore.containsKey(key), is(false));
+    if(validException) {
```

Comment:
I think that using a boolean and throw the exception lazily is dangerous, if an exception is thrown, due to a failure, even valid, it is likely that the assertions are going to fail thus marking the test as fail while it would be valid to mark it as passing.
This comment apply to all other test cases that use a boolean